server/src/test/java/org/apache/iotdb/db/writelog/io/LogWriterReaderTest.java

```diff
@@ -94,4 +97,102 @@ public void testWriteAndRead() throws IOException {
       new File(filePath).delete();
     }
   }
+
+  @Test
+  public void testReachEOF() throws IOException {
+    try {
+      // write normal data
+      LogWriter writer =
+          new LogWriter(
+              filePath, IoTDBDescriptor.getInstance().getConfig().getForceWalPeriodInMs() == 0);
+      writer.write(logsBuffer);
+      writer.force();
+      writer.close();
+      long expectedLength = new File(filePath).length();
+
+      // just write partial content
+      FileChannel channel = new FileOutputStream(filePath, true).getChannel();
+      ByteBuffer logBuffer = ByteBuffer.allocate(4 * 30);
+      for (int i = 0; i < 20; ++i) {
+        logBuffer.putInt(Integer.MIN_VALUE);
+      }
+      logBuffer.flip();
+      ByteBuffer lengthBuffer = ByteBuffer.allocate(4);
+      lengthBuffer.putInt(logBuffer.capacity());
+      lengthBuffer.flip();
+
+      channel.write(lengthBuffer);
+      channel.write(logBuffer);
+      channel.force(true);
+      channel.close();
+
+      // read & check
+      SingleFileLogReader reader = new SingleFileLogReader(new File(filePath));
+      List<PhysicalPlan> res = new ArrayList<>();
+      while (reader.hasNext()) {
+        res.add(reader.next());
+      }
+      for (int i = 0; i < plans.size(); i++) {
+        assertEquals(plans.get(i), res.get(i));
+      }
+      reader.close();
+      assertEquals(expectedLength, new File(filePath).length());
+    } finally {
+      new File(filePath).delete();
```

Comment:
please move "channel.close(), reader.close()"  to "finally". 
If the test case is abnormal and the connection is not closed, the execution of other test cases will be affected.

Reply:
Fixed.